docs/api/services.md

```diff
@@ -0,0 +1,105 @@
+.. cpp:function:: bool bm_service_register(size_t service_strlen, const char *service, \
+                                           BmServiceHandler service_handler)
+
+  Register a service.
+
+  :param service_strlen: The length of the service string
+  :param service: The service string
+  :param service_handler: The callback to call when a request is received
+
+  :returns: true if the service was registered, false otherwise
+```
+
+### Unregistering Services
+
+To unregister a service, you must provide the service ID of the service you wish to unregister.
+Unregistering a service will remove the service from the list of available services and service
+requests will no longer be forwarded to the service handler callback.
+
+```{eval-rst}
+.. cpp:funciton:: bool bm_service_unregister(size_t service_strlen, const char *service)
+
+  Unregister a service.
+
+  :param service_strlen: The length of the service string
+  :param service: The service string
+
+  :returns: true if the service was unregistered, false otherwise
+```
+
+## Service Requests
+
+The service requests (`service_request.h\.c`) is responsible for tracking outbound service requests
+and invoking the appropriate service response callback when a response is received.
+Outbound requests each should have a unique ID, callback, and timeout.
+If the request times out, the service request handler will invoke the callback with a "fail".
+If the request is successful, the service request handler will invoke the callback with a "success".
+
+When a service request is made, the request data will be sent to the service on the topic
+`node_id/service ID/req`, the topic that the recipient should be subscribed to.
+This topic will then appear in the requesters's publication list.
+Likewise, the requester will subscribe to the topic `node_id/service ID/rep` to receive the response.
+This topic will appear in the requester's subscription list.
+
+### Sending Service Requests
+
+To send a service request, you must provide the service ID of the service you wish to request,
+the data that may be sent as part of the request, a callback to be invoked when the reply is received,
+and a timeout to expire the request if no reply is received.
+
+```{eval-rst}
+.. cpp:function:: bool bm_service_request(size_t service_strlen, const char *service, \
+                                          size_t data_len, const uint8_t *data, \
+                                          BmServiceReplyCb reply_cb, uint32_t timeout_s)
+
+  Send a service request.
+
+  :param service_strlen: The length of the service string
+  :param service: The service string
+  :param data_len: The length of the data
+  :param data: The data to send as part of the request
+  :param reply_cb: The callback to call when a reply is received
+  :param timeout_s: The timeout in seconds
+
+  :returns: true if the request was sent, false otherwise
+```
+
+## Examples
+
+### echo_service
+<!-- TODO -->
+
+### sys_info_service
+<!-- TODO -->
```

Comment:
Do we want to include examples here? Save that for later?

Reply:
I think yeah, it would be great to give a super brief example here inline using echo, and then link to the code for sys info with language like, "You can find a more comprehensive example in the `sys_info_service`."

Reply:
@victorsowa12 these are already described in the `middleware.md` file and might not be necessary to include here

Reply:
Okay, I can add a link to go checkout the middleware.md for examples